packages/labs/react/src/create-component.ts

```diff
@@ -84,16 +84,34 @@ const setRef = (ref: React.Ref<unknown>, value: Element | null) => {
   }
 };
 
-type Events<S> = {
-  [P in keyof S]?: (e: Event) => unknown;
-};
-
 type StringValued<T> = {
   [P in keyof T]: string;
 };
 
 type Constructor<T> = {new (): T};
 
+type EventString<T = unknown> = string & {
+  __event_type: T;
+};
+
+/***
+ * Typecast that curries an Event type through an EventString. The React
+ * equivelant to an event listeners is an `EventHandler`. The goal of the type
+ * name is to hint that corresponding EventHandler callback in their props will
+ * be of a certain type.
+ */
+export type EventHandler<T = unknown> = T extends Event
```

Comment:
I don't think you need this, especially if you constrain the type parameter in `EventString`.

Reply:
Done :)